crates/supervisor/storage/src/models/mod.rs

```diff
@@ -69,6 +76,22 @@ tables! {
         type Key = u64;
         type Value = BlockRef;
     }
+
+    /// A table mapping a derived block number to its corresponding source and derived block reference.
+    /// - Key: `u64` — derived block number
+    /// - Value: [`DerivedBlockPair`] — pair of source and derived block reference
+    table DerivedBlocks {
+        type Key = u64;
+        type Value = DerivedBlockPair;
+    }
+
+    /// A table mapping a source block number to a list of its derived block numbers.
+    /// - Key: `u64` — source block number
+    /// - Value: [`U64List`] — list of derived block numbers
+     table SourceToDerivedBlockNumbers {
+        type Key = u64;
+        type Value = U64List;
+    }
 }
```

Comment:
Instead of maintaining two table we can create a DUP SORT table like this:
```
table DerivedBlocks {
        type Key = u64; // source block
        type SubKey = u64 // derived block
        type Value = DerivedBlockPair;
    }
```

Reply:
We need to have both lookup to fetch derived blocks from source block and source block by derived block. I don't think we can achieve this by `DUP SORT` table